core-bundle/src/DependencyInjection/Compiler/SearchIndexerPass.php

```diff
@@ -28,31 +29,36 @@ class SearchIndexerPass implements CompilerPassInterface
 
     public function process(ContainerBuilder $container): void
     {
-        if (!$container->has(self::DELEGATING_SERVICE_ID)) {
-            return;
-        }
-
-        $definition = $container->findDefinition(self::DELEGATING_SERVICE_ID);
-        $references = $this->findAndSortTaggedServices('contao.search_indexer', $container);
+        $indexers = $this->findAndSortTaggedServices('contao.search_indexer', $container);
 
         // Make sure we do not add the delegating indexer to itself to prevent endless redirects
-        $references = array_filter(
-            $references,
+        $indexers = array_filter(
+            $indexers,
             static function (Reference $reference): bool {
                 return self::DELEGATING_SERVICE_ID !== (string) $reference;
             }
         );
 
-        // Remove the service and the search index listener if there are no indexers
-        if (0 === \count($references)) {
+        if (!$container->hasDefinition(self::DELEGATING_SERVICE_ID) || 0 === \count($indexers)) {
+            // Remove delegating indexer
             $container->removeDefinition(self::DELEGATING_SERVICE_ID);
```

Comment:
Same here. Is this intentional?

Reply:
Yes it is. It's either not there completely or there are no indexers/crawl subscribers. It's just abbreviated :)